### ISSUE TITLE
Assertions of keyed arrays

### ISSUE THREAD

**Issue by zerkms** [vimeo/psalm#2067]
https://psalm.dev/r/eea47439ea

```php
<?php
  
/**
 * @psalm-assert array{id: string} $v
 */
function foo($v) : void {}
```
Expected: all good

Actual: 
```
ERROR: MissingDocblockType - 6:1 - Misplaced variable in docblock for foo

INFO: MissingParamType - 6:14 - Parameter $v has no provided type
```

**Comment by weirdan** [comment]
It actually kinda works (apart from spurious RedundantCondition): https://psalm.dev/r/2dd46bf851

**Comment by muglug** [comment]
OH it's just the space that's the problem

**Comment by muglug** [comment]
You can get ride of the redundant condition by prefixing with a `=`: https://psalm.dev/r/44bb97c16f

**Comment by weirdan** [comment]
> You can get ride of the redundant condition by prefixing with a =
It's still quite unexpected, as `$_GET` shape should be unknown at that point.

**Comment by muglug** [comment]
`$_GET` is known to be an array, though

**Comment by weirdan** [comment]
> `$_GET` is known to be an array, though

But `$_GET['a']` is unknown, and still results in the same error reported: https://psalm.dev/r/40fc8c86a0

**Comment by muglug** [comment]
Yeah, agreed

**Comment by muglug** [comment]
Fixed in 0ae04c9
